### ISSUE TITLE
Support requestInterceptor & responseInterceptor on a per-call basis using tag interface

### ISSUE THREAD

**Issue by matthewadams** [swagger-api/swagger-js#1290]
In `swagger-client@3.7.0`, you can only set a `requestInterceptor` or `responseInterceptor` for all calls via the `Swagger()` function, which we use for customization of HTTP headers.  However, some of the headers we are required to send differ by endpoint.  For example, all calls may need to set header `X-Foo: Bar`, which we can set via the `Swagger()` function, but endpoint `POST /snafu` requires additional header `X-Snafu: bork`, while enpoint `GET /snort` requires additional header `X-Wonk: chordle`.  If the generated tag interface methods supported `requestInterceptor` & `responseInterceptor` properties, we could handle this requirement easily.

Further, if interceptors are specified at _both_ the top level (via the `Swagger()` function) _and_ when calling a tag interface method, the top-level interceptor should be called first, then the call-level interceptor should be called.  This allows call-level interceptors to override modifications made by the top-level interceptors.

**Comment by vbyjsue** [comment]
I've just recently begun using this lib so I might be wrong, but this works for me in 3.7.2:
```
client.apis.user.get_user_session(
{},
{requestInterceptor: req => { console.log(req); return req;}}
)
```

**Comment by char0n** [comment]
@vbyjsue thanks for suggesting the possible solution to this.

**Comment by char0n** [comment]
@matthewadams although I do agree, that SwaggerClient could do implicit composition of interceptors, introducing it now would  possibly introduce a breaking behavior change for existing library consumers. 

I'm also a concerned about the consistency of the API. Second argument to every operation from [Tags Interface](https://github.com/swagger-api/swagger-js/blob/master/docs/usage/tags-interface.md) is an `override` mechanism. `requestInterceptors` and `responseInterceptor` would be then exception to this rule and there will be no mechanism to override/drop the top level interceptor. 

So in a light of those argument I do think current API is more flexible and allows more use-cases. I've added your usecase to our documentation via #1619. 

Thank you for bringing this issue up!

Closing this issue as we now have a documented solution.
